### PR TITLE
Feature/v2 updates

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**Are you a current Fivetran customer?** 

Fivetran created PR

**What change(s) does this PR introduce?** 

This PR applies the Ad Reporting V2 updates. In particular this PR does the following:
- Adds the `ad_group_criterion_history` source and accompanying staging models
- Adds the `ad_group_stats` source and accompanying staging models
- Adds the `campaign_stats` source and accompanying staging models
- Adds the `account_stats` source and accompanying staging models
- Adds the `keyword_stats` source and accompanying staging models
- Adds tests to each of the new and existing end models
- Completely removes the Adwords versions of the package as it has been fully sunset
- Applies README updates
- Adds identifiers to all source models

**Did you update the CHANGELOG?** 

- [ ] Yes

**Does this PR introduce a breaking change?**

- [X] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)

This will be a major update and will be breaking for existing users.

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [X] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [ ] Yes, Issue/Feature [link bug/feature number here]
- [X] No 

**How did you test the PR changes?** 

 

- [X] CircleCi  
- [X] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [X] BigQuery
- [X] Redshift
- [X] Snowflake
- [X] Postgres
- [X] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
✌️ 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

**Comment by fivetran-joemarkiewicz** [comment]
FYI the base branch is `feature/google-ads-api-default` as that branch will be merged prior to this one and I want to make sure I am not duplicating reviews from the changes made in PR #28

File: models/stg_google_ads__account_history.sql
Comment (fivetran-sheringuyen) [inline models/stg_google_ads__account_history.sql]: I'm interested in your preference for `updated_at` vs `updated_timestamp` and whether you think this is worth noting in our style guide. 🤔

File: models/stg_google_ads__account_stats.sql
Comment (fivetran-sheringuyen) [inline models/stg_google_ads__account_stats.sql]: I faintly remember a discussion the team had about the usage of `var('google_ads__account_stats_passthrough_metrics', [])` vs more explicit like the use of lengths or something similar, but I can't remember which package that new logic was in. I think it was actually a package that you created a PR for but I reviewed it..does this ring a bell? (I should have definitely noted this better somewhere 😅 )

File: CHANGELOG.md
Comment (fivetran-sheringuyen) [inline CHANGELOG.md]: Is there a preference of referring to the same PR multiple times within the `CHANGELOG` vs outlining just once at the top which PR all the changes were made in?

File: README.md
Comment (fivetran-sheringuyen) [inline README.md:19]: Maybe "Google Ads"?

File: README.md
Comment (fivetran-sheringuyen) [inline README.md:17]: I forgot to add this too -- but it looks like there is currently no freshness test for the Google source data. 

For the Apple Store & Google Play package, we used the below:
>     freshness: 
>       warn_after: {count: 48, period: hour}
>       error_after: {count: 168, period: hour}

Would you say this is also a good freshness standard for our Ad Reporting as well or would you suggest something different?

File: models/src_google_ads.yml
Comment (fivetran-sheringuyen) [inline models/src_google_ads.yml:2]: Add freshness tests

File: README.md
Comment (fivetran-sheringuyen) [inline README.md]: We could probably remove this comment now!

File: README.md
Comment (fivetran-sheringuyen) [inline README.md]: Maybe instead of "google_ads" we use "Google Ads"?

> By default, this package builds the Google Ads staging models within a schema titled (`<target_schema>` + `_stg_google_ads`) in your destination. If this is not where you would like your Google Ads staging data to be written to, add the following configuration to your root `dbt_project.yml` file:

File: README.md
Comment (fivetran-sheringuyen) [inline README.md]: > If this is not where your Google Ads data is ...

**Review by fivetran-sheringuyen — COMMENTED** [review]
Hey @fivetran-joemarkiewicz, great job on the deprecation and rest of the PR! I just had some notes I wanted us to keep track of and a couple in-line comments.

There may be some merge conflicts since the `feature/google-ads-api-default` updates, but I want to make sure certain things don't accidentally get lost when we do get around to releasing this package:
- [x] Documentation for final urls regex matching - specifically in the `stg_google_ads__ad_history.source_final_urls` tests.
- [x] Removing deprecation notes from `README`

It looks like in both `google_ads` and `google_ads_source`, `run_results.json` is missing. Is this intentional (and should be updated in the `generate_docs` macro)?

File: dbt_project.yml
Comment (fivetran-sheringuyen) [inline dbt_project.yml]: +schema: google_ads_source (to reduce redundancy)

File: README.md
Comment (fivetran-sheringuyen) [inline README.md]: With changing the redundancy:

> <target_schema> + `_google_ads_source`

File: models/stg_google_ads__account_stats.sql
Comment (fivetran-joemarkiewicz) [reply models/stg_google_ads__account_stats.sql]: Good call out! That conversation was regarding our use of adding passthrough columns with an `if` statement. 
https://github.com/fivetran/dbt_zendesk_source/blob/aef4f3e6d436b4d5f52ecac81c8beb71951289b0/models/stg_zendesk__ticket.sql#L25-L30

I actually did not realize we also use the `for loop` version until I came across our use of them within these ad packages. Personally, I like the `for loop` better as it reads a bit more naturally and is explicit by default (imo). The `if` statement is pretty clunky and hard to follow, whereas the `for loop` naturally tells me to iterate through the variable without needing a condition and just sets a default to be safe.

I would like to actually suggest we start using the for loop method from now on.

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [reply CHANGELOG.md]: Yeah I saw that you highlighted all of the changes were included in the single PR for your Changelog. I think that is a good call. When there are multiple PRs or just one minor update that is tied to a single PR then we can take the approach I have done here. 

I will update my CHANGELOG to match that style.

File: README.md
Comment (fivetran-joemarkiewicz) [reply README.md:17]: Yeah I think that is a good standard! Let's use that going forward and I will add the tests to this package.

Comment (fivetran-joemarkiewicz) [reply README.md:17]: Updated

File: models/src_google_ads.yml
Comment (fivetran-joemarkiewicz) [reply models/src_google_ads.yml:2]: Added

File: README.md
Comment (fivetran-joemarkiewicz) [reply README.md]: Removed

File: README.md
Comment (fivetran-joemarkiewicz) [reply README.md]: Updated

File: README.md
Comment (fivetran-joemarkiewicz) [reply README.md]: Updated

File: dbt_project.yml
Comment (fivetran-joemarkiewicz) [reply dbt_project.yml]: Updated

File: README.md
Comment (fivetran-joemarkiewicz) [reply README.md]: Updated

File: README.md
Comment (fivetran-joemarkiewicz) [reply README.md:19]: Updated

File: CHANGELOG.md
Comment (fivetran-joemarkiewicz) [reply CHANGELOG.md]: Updated

File: models/stg_google_ads__account_history.sql
Comment (fivetran-joemarkiewicz) [reply models/stg_google_ads__account_history.sql]: It looks like this is the only one of our packages that does this `_timestamp` name change. I just switched back to just use the `_at` convention to be consistent with our other packages. Additionally, I am not entirely sold on the requirement of this naming change.

Comment (fivetran-joemarkiewicz) [reply models/stg_google_ads__account_history.sql]: Made the change to the models and .yml documentation.

**Comment by fivetran-joemarkiewicz** [comment]
Thanks for the review @fivetran-sheringuyen! I addressed your inline comments and only had a question on the one I left unresolved regarding the passthrough logic. Let me know your thoughts! Additionally I addressed the following that was not captured inline:
- Added the yml documentation for the test warning with the `stg_google_ads__ad_hsitory` final_urls field.
- Included the `run_results.json`. I initially intentionally omitted it as it isn't necessarily needed. However, I noticed the datatype value is missing within the docs if you omit this file. Therefore I added it back in.
- Addressed merge conflicts after switching the base branch to be `main`.
- Regenerated the docs.

File: models/stg_google_ads__account_stats.sql
Comment (fivetran-sheringuyen) [reply models/stg_google_ads__account_stats.sql]: I agree, I also like the for loop method better! Will add this to the dev philo. Thanks Joe!

**Review by fivetran-sheringuyen — APPROVED** [review]
Thanks @fivetran-joemarkiewicz for making the updates, this package looks good to me for release when we roll out Ad Reporting V2!

**Comment by fivetran-joemarkiewicz** [comment]
Things to update before merging:
- [x] Add not null tests
- [x] Ensure passthrough metrics is correct in both CTEs and no duplicates in macro

**Pull request merged**
